envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyClustersFactory.kt

```diff
@@ -73,13 +73,20 @@ internal class EnvoyClustersFactory(
         return when (group) {
             is ServicesGroup -> group.proxySettings.outgoing.getServiceDependencies()
                 .mapNotNull {
-                    if (enableTlsForCluster(group, globalSnapshot, it.service)) {
-                        globalSnapshot.securedClusters.resources().get(it.service)
-                    } else {
-                        globalSnapshot.clusters.resources().get(it.service)
-                    }
+                    selectCluster(group, globalSnapshot, it.service)
+                }
+            is AllServicesGroup -> globalSnapshot.allServicesGroupsClusters
+                .mapNotNull { (service, _) ->
+                    selectCluster(group, globalSnapshot, service)
                 }
-            is AllServicesGroup -> globalSnapshot.allServicesGroupsClusters.map { it.value }
```

Comment:
We can degrade performance this way, because until now we had a predefined list of clusters for services with wildcard.
Now we will unnecessary calculate list of clusters for every wildcard group. It is unnecessary, because in reality we will have only two sets of clusters:
1. Clusters for wildcard, mtls enabled groups
2. Clusters for wildcard, mtls disabled groups

So we can precalculate these two sets of clusters, instead of calculating it for every group. But I think it will complicate the code, so I would test if your approach affects performance (with many wildcard services). If it is not, we can go with this approach IMO.

Reply:
It's 3 map lookups + one `if` I will measure it.

Reply:
It's `1000 x <number of groups with wildcard>` map lookups per every Snapshot